pkg/admissioncontroller/webhook/admission/authenticationconfig/handler.go

```diff
@@ -118,18 +123,26 @@ func (h *Handler) admitShoot(ctx context.Context, request admission.Request) adm
 
 		oldShootKubernetesVersion = oldShoot.Spec.Kubernetes.Version
 		oldAuthenticationConfigurationConfigMapName = gardencorehelper.GetShootAuthenticationConfigurationConfigMapName(oldShoot.Spec.Kubernetes.KubeAPIServer)
+		oldServiceAccountIssuer = gardencorehelper.GetShootServiceAccountConfigIssuer(oldShoot.Spec.Kubernetes.KubeAPIServer)
+		oldServiceAccountAcceptedIssuers = gardencorehelper.GetShootServiceAccountConfigAcceptedIssuers(oldShoot.Spec.Kubernetes.KubeAPIServer)
 	}
 	newShootKubernetesVersion = shoot.Spec.Kubernetes.Version
 	newAuthenticationConfigurationConfigMapName = gardencorehelper.GetShootAuthenticationConfigurationConfigMapName(shoot.Spec.Kubernetes.KubeAPIServer)
+	newServiceAccountIssuer = gardencorehelper.GetShootServiceAccountConfigIssuer(shoot.Spec.Kubernetes.KubeAPIServer)
+	newServiceAccountAcceptedIssuers = gardencorehelper.GetShootServiceAccountConfigAcceptedIssuers(shoot.Spec.Kubernetes.KubeAPIServer)
 
 	if newAuthenticationConfigurationConfigMapName == "" {
 		return admissionwebhook.Allowed("shoot resource is not specifying any authentication configuration")
 	}
 
 	// oldAuthenticationConfigurationConfigMapName is empty for CREATE shoot requests that specify authentication configuration reference
 	// if Kubernetes version is changed we need to revalidate if the authentication configuration API version is compatible with
-	// new Kubernetes version
-	if oldAuthenticationConfigurationConfigMapName == newAuthenticationConfigurationConfigMapName && oldShootKubernetesVersion == newShootKubernetesVersion {
+	// new Kubernetes version. We also need to revalidate when the ServiceAccountConfig allowed issuers have changes since they are
+	// disallowed for the authentication configuration
+	if oldAuthenticationConfigurationConfigMapName == newAuthenticationConfigurationConfigMapName &&
+		oldShootKubernetesVersion == newShootKubernetesVersion &&
+		equalOrNil(oldServiceAccountIssuer, newServiceAccountIssuer) &&
+		slices.Equal(oldServiceAccountAcceptedIssuers, newServiceAccountAcceptedIssuers) {
```

Comment:
This check already seems complicated enough. Isn't a more practical approach to always validate the configuration 🤔

Reply:
Then number of checks here is getting a bit too much. I have now removed then and the validation is now happening on every shoot Update.